### PR TITLE
doxygen: Move doxygen files into docs

### PR DESCRIPTION
**Pull request by jdemel**
This is a first stab at re-arranging doxygen files to have fewer files in our top-level directory.

This includes all `*.md` files as well. We might want to exclude some of them.

### CODE REVIEW

File: CMakeLists.txt
Comment (michaelld) [inline CMakeLists.txt]: Let's move this whole section, from `find_package(Doxygen)` to `endif(DOXYGEN_FOUND)` into the subdirectory. That's better CMake programming: only check in the current CMakeLists.txt file for stuff that's required within it, and don't rely on previous checks that may or not have taken place.

**Review by michaelld — CHANGES_REQUESTED** [review]
Minor tweak & this will be good to go. I need to learn how to edit PRs ... not tonight though

File: CMakeLists.txt
Comment (jdemel) [reply CMakeLists.txt]: I agree. Are there actually any resources that teach you "good CMake"?

**Comment by jdemel** [comment]
Should some of the `*.md` files be excluded?

File: CMakeLists.txt
Comment (michaelld) [reply CMakeLists.txt]: Hmmm ... good question ... IDK .. mine is via experience, having used CMake now since GR switched, ages ago; and, working on tons or projects that use CMake

**Comment by michaelld** [comment]
> Should some of the `*.md` files be excluded?

Not sure I follow this comment for this PR? Maybe you meant #580 ?

**Comment by jdemel** [comment]
If we'd merge #580 (and I assume we do now), then my comment is more relevant. I'll wait for the other things to happen and update this PR accordingly. I assume I mixed up some files.

**Comment by michaelld** [comment]
OK #580 is merged ... once this PR's code is tweaked per my request then I'm good with it. @jdemel if there are other changes needed post-580 then please go ahead & push commits for them for us to review.

File: CMakeLists.txt
Comment (jdemel) [reply CMakeLists.txt]: I moved it.

**Comment by jdemel** [comment]
Some `.md` files that I consider to be clutter are included in the docs at the moment.

I feel like these should not be included:
- top-level `README.md`
- docs/AUTHORS_RESUBMITTING_UNDER_LGPL_LICENSE.md`

e.g. `docs/versioning.md` were not part of our docs build before. But I assume these fit nicely in the docs as well.

**Review by michaelld — APPROVED** [review]
LGTM now; thx!

**Comment by michaelld** [comment]
The top-level `README.md` is required for best GitHub presentation, so that needs to stay there. I'm OK with `AUTHORS_RESUBMITTING_UNDER_LGPL_LICENSE.md` and `versioning.md` being in `docs` since they fit there. Once this PR is merged, I think we're at a minimum number of extraneous files at the top-level ... do you agree?

**Comment by michaelld** [comment]
I mean those extra `.md` files in `docs` are just along for the ride; we don't build them into the actual Doxygen docs beyond links, do we?

**Comment by michaelld** [comment]
If you feel some of those files need to be moved & it makes sense to put them into this PR to go for it. Otherwise, I'd say let's merge this one & get the next round going.

**Comment by jdemel** [comment]
I want these files to stay where they are. I just don't want to integrate them into the Doxygen build. I'll exclude them.

**Comment by jdemel** [comment]
Now, this PR should be ready to be merged. @michaelld I leave this to you since it is my PR.

**Review by michaelld — APPROVED** [review]
LGTM ; merging

**Pull request merged**
